packages/react/src/components/accordion/Accordion.tsx

```diff
@@ -119,21 +179,54 @@ export const Accordion = ({
     <IconAngleDown aria-hidden className={styles.accordionButtonIcon} />
   );
 
+  const didMount = useRef(false);
+
+  /*
+    Close button does not work well with different screen readers. To avoid issues, we had to:
+    1. Add aria-expanded to false before closing the accordion (state beforeCloseButtonClick)
+    2. Wait for the new rendering (useEffect + timeout)
+    3. Focus to the accordion main button to not lose focus when close button disappears
+    4. Finally call the regular button onClick, which updates state and hides the content.
+   */
+
+  useEffect(() => {
+    if (!didMount.current) {
+      didMount.current = true;
+      return;
+    }
```

Comment:
Should we add an internal hook to detected mounts?

```
/*
  Simple hook that will let us know if a
  component has mounted yet.
  Doesn't track unmounts.
*/
import React from 'react';
function useHasMounted() {
  const [hasMounted, setHasMounted] = React.useState(false);
  React.useEffect(() => {
    setHasMounted(true);
  }, []);
  return hasMounted;
}
export default useHasMounted;
```

Reply:
- [ ] Use useHasMounted hook